editor/components/AddressFormField.tsx

```diff
@@ -0,0 +1,90 @@
+import { Form } from "react-bootstrap";
+import { useLookupQuery } from "@/utils/graphql";
+import { useQuery } from "@/utils/graphql";
+import { LookupTableOption } from "@/types/relationships";
+import { AddressFormInputs } from "@/components/EditCrashAddressModal";
+import { Relationship } from "@/types/relationships";
+import { UseFormRegister } from "react-hook-form";
+import { Spinner } from "react-bootstrap";
+
+interface AddressFormFieldProps {
+  /**
+   * These props are pulled from the crashesColumns config
+   */
+  col: {
+    path: keyof AddressFormInputs; // Ensures the path matches one of the AddressInputs keys
+    label: string;
+    inputType: "text" | "number" | "select";
+    relationship?: Relationship<Record<string, unknown>>; // Optional since it's conditionally used
+  };
+  /**
+   * This method comes from RHF useForm()
+   */
+  register: UseFormRegister<AddressFormInputs>;
+}
+
+/**
+ * Component for the address modal that that renders a reach hook form field based on the input type
+ * of the column, such as text or select
+ */
+export default function AddressFormField({
+  col,
+  register,
+}: AddressFormFieldProps) {
+  const [query, typename] = useLookupQuery(
+    col?.relationship ? col.relationship : undefined
+  );
+
+  const {
+    data: selectOptions,
+    error: selectOptionsError,
+    isLoading: isLoadingSelectOptions,
+  } = useQuery<LookupTableOption>({
+    query,
+    // we don't need to refetch lookup table options
+    options: { revalidateIfStale: false },
+    typename,
+  });
+  const inputType = col.inputType;
+
+  return (
+    <div className="mb-2">
+      {(inputType === "text" || inputType === "number") && (
+        <Form.Control
+          {...register(col.path, {
+            setValueAs: (value) => value?.trim() || null,
+            onChange: (e) => (e.target.value = e.target.value.toUpperCase()), // uppercase input as user is typing
```

Comment:
i made the executive decision to just uppercase user input as they are typing bc its gonna get save like that anyways so i think it makes sense for users to see whats going to get saved :shrug: